### PR TITLE
Hypixel friend

### PR DESCRIPTION
**Pull request by unpluggedsam**

### CODE REVIEW

**Review by TheNullicorn — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/hypixel_api_wrapper/wrapper/player/HypixelFriend.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import java.time.Instant;
+
+/**
+ * This class is used to link two {@link HypixelPlayer}'s as friends on the Hypixel network.
+ *
+ * @author sam & insert
+ */
+public class HypixelFriend {
+
+    private final HypixelPlayer player;
```
[the remark this answers is not in the document]
Comment (unpluggedsam) [reply src/main/java/io/github/hypixel_api_wrapper/wrapper/player/HypixelFriend.java:12]: @NotInsertIGN and I had this conversation. The purpose of having a reference to the player is not to have a getter (hence the reason there is no getter for it) but for the getDateAdded() method. It needs to know both players to check the date they added each other as friends, or else there would be no point for this class.

File: src/main/java/io/github/hypixel_api_wrapper/wrapper/player/HypixelFriend.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    private final HypixelPlayer friend;
+    private final String apiKey;
+
+    public HypixelFriend(HypixelPlayer player, HypixelPlayer friend, String apiKey) {
+        this.player = player;
+        this.friend = friend;
+        this.apiKey = apiKey;
+    }
+
+    public HypixelPlayer getFriend() {
```
[the remark this answers is not in the document]
Comment (unpluggedsam) [reply src/main/java/io/github/hypixel_api_wrapper/wrapper/player/HypixelFriend.java:22]: The purpose of this class is a way to link the two HypixelPlayer's together and the date they added each other as a friend. So, for example `HypixelFriend friend = player.getFriend(username)` would return a `HypixelFriend`. You could then call `friend.getDateAdded()` or `friend.getFriend()`. I see how it is repetitive I'm just can't really think of a better name.

[the remark this answers is not in the document]
Comment (TheNullicorn) [reply src/main/java/io/github/hypixel_api_wrapper/wrapper/player/HypixelFriend.java:22]: Maybe `toPlayer()` or `asPlayer()`?

[the remark this answers is not in the document]
Comment (unpluggedsam) [reply src/main/java/io/github/hypixel_api_wrapper/wrapper/player/HypixelFriend.java:22]: `asPlayer` sounds good.

File: src/main/java/io/github/hypixel_api_wrapper/wrapper/player/HypixelFriend.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheNullicorn) [reply src/main/java/io/github/hypixel_api_wrapper/wrapper/player/HypixelFriend.java:12]: Oh okay, that works. If you don't mind, a TODO comment above that field would help clarify that until it's implemented

[the remark this answers is not in the document]
Comment (unpluggedsam) [reply src/main/java/io/github/hypixel_api_wrapper/wrapper/player/HypixelFriend.java:12]: Will do

**Review by unpluggedsam — COMMENTED** [review]
Apologies, I didn't see the reviews and ended up making the commits myself.

**Review by TheNullicorn — APPROVED** [review]

**Pull request merged**